packages/dds/tree/src/feature-libraries/chunked-forest/chunkedForest.ts

```diff
@@ -158,6 +159,7 @@ export class ChunkedForest implements IEditableForest {
 			 * If not specified, the detached range is destroyed.
 			 */
 			detachEdit(source: Range, destination: FieldKey | undefined): void {
+				this.forest.events.emit("beforeChange");
```

Comment:
Having it on detachEdit() and attachEdit() will make it fire twice during replace(). Would it be better to have it on detac(), attach(), and replace()? (destroy can keep it). Same for objectForest.

If so, probably worth a test to make sure replace() behaves correctly (at least for replace() since given the existing code patterns it seems easier for that one to slip compared to the others, although I might check all of them the same way).